File: src/day1.py
def main():
    calories = []
    tmp = []
    with open("src/day1") as file:
        data = file.readlines()
        for line in data:
            if line == "\n":
                calories.append(sum(tmp))
                tmp = []
            else:
                tmp.append(int(line.replace("\n", "")))


        print(calories)

    max_value = max(calories)
    max_index = calories.index(max_value)
    print(max_value, max_index)

    calories.sort()

    print(sum(calories[-3:]))
    print(f"Solution: {max_value} and {sum(calories[-3:])}")

if __name__ == '__main__':
    main()

File: src/main/kotlin/aoc2022/day1.kt
package aoc2022

import java.io.File

fun main() {
    val fileName = "src/main/resources/aoc2022/day1"
    var lines: List<String> = File(fileName).bufferedReader().readLines()
    println(lines)
    var calories = mutableListOf<Int>()
    var tmp = mutableListOf<Int>()

    for (line in lines) {
        println(line)
        if (line == "\n" || line.isEmpty()) {
            calories += tmp.sum()
            tmp = mutableListOf()
        } else
            tmp += line.replace("\n", "").toInt()
    }
    calories.sort()
    println(calories)
    println(calories.max())
    println(calories.slice(calories.size - 3..calories.size -1).sum())

}